query-dsl/src/main/java/org/infinispan/query/dsl/impl/BaseQueryBuilder.java

```diff
@@ -51,29 +58,74 @@ protected String getRootTypeName() {
    }
 
    @Override
-   public QueryBuilder<T> orderBy(String attributePath, SortOrder sortOrder) {
+   public QueryBuilder<T> orderBy(Expression pathExpression, SortOrder sortOrder) {
       if (sortCriteria == null) {
          sortCriteria = new ArrayList<SortCriteria>();
       }
-      sortCriteria.add(new SortCriteria(attributePath, sortOrder));
+      sortCriteria.add(new SortCriteria(pathExpression, sortOrder));
       return this;
    }
 
+   @Override
+   public QueryBuilder<T> orderBy(String attributePath, SortOrder sortOrder) {
+      return orderBy(Expression.property(attributePath), sortOrder);
+   }
+
    protected List<SortCriteria> getSortCriteria() {
       return sortCriteria;
    }
 
    @Override
-   public QueryBuilder<T> setProjection(String... projection) {
+   public QueryBuilder<T> select(String... attributePath) {
+      if (attributePath == null || attributePath.length == 0) {
+         throw new IllegalArgumentException("Projection cannot be null or empty");
```

Comment:
Some of these seem like programming errors, which I wouldn't expect an externalized log or do we want to externalize those cases as well?